WordPress/Classes/Services/SharingService.swift

```diff
@@ -0,0 +1,413 @@
+    */
+    public func deletePublicizeConnection(pubConn:PublicizeConnection, success: (() -> Void)?, failure: (NSError! -> Void)?) {
+        // optimistically delete the connection locally.
+        let siteID = pubConn.siteID;
+        managedObjectContext.deleteObject(pubConn);
+        ContextManager.sharedInstance().saveContext(managedObjectContext)
+
+        let remote = SharingServiceRemote(api: apiForRequest())
+        remote.deletePublicizeConnection(siteID, connectionID:pubConn.connectionID, success:success, failure:failure)
+    }
+
+
+    // MARK: - Public PublicizeService Methods
+
+    /**
+    Finds a cached `PublicizeService` matching the specified service name.
+
+    @param name The name of the service. This is the `serviceID` attribute for a `PublicizeService` object.
+
+    @return The requested `PublicizeService` or nil.
+    */
+    public func findPublicizeServiceNamed(name:String) -> PublicizeService? {
+        let request = NSFetchRequest(entityName: PublicizeService.classNameWithoutNamespaces())
+        request.predicate = NSPredicate(format: "serviceID = %@", name)
+
+        var services: [PublicizeService]
+        do {
+            services = try managedObjectContext.executeFetchRequest(request) as! [PublicizeService]
+        } catch let error as NSError {
+            DDLogSwift.logError("Error fetching Publicize Service named \(name) : \(error.localizedDescription)")
+            services = []
+        }
+
+        return services.first
+    }
+
+
+    /**
+    Returns an array of all cached `PublicizeService` objects.
+
+    @return An array of `PublicizeService`.  The array is empty if no objects are cached.
+    */
+    public func allPublicizeServices() -> [PublicizeService] {
+        let request = NSFetchRequest(entityName: PublicizeService.classNameWithoutNamespaces())
+        let sortDescriptor = NSSortDescriptor(key: "order", ascending: true)
+        request.sortDescriptors = [sortDescriptor]
+
+        var services: [PublicizeService]
+        do {
+            services = try managedObjectContext.executeFetchRequest(request) as! [PublicizeService]
+        } catch let error as NSError {
+            DDLogSwift.logError("Error fetching Publicize Services: \(error.localizedDescription)")
+            services = []
+        }
+
+        return services
+    }
+
+
+    // MARK: - Private PublicizeService Methods
+
+
+    /**
+    Called when syncing Publicize services. Merges synced and cached data, removing
+    anything that does not exist on the server.  Saves the context.
+
+    @param remoteServices An array of `RemotePublicizeService` objects to merge.
+    @param success An optional callback block to be performed when core data has saved the changes.
+    */
+    private func mergePublicizeServices(remoteServices:[RemotePublicizeService], success:(() -> Void)? ) {
+        managedObjectContext.performBlock { () -> Void in
```

Comment:
Are you running `lint` on the CLI? (i'm not getting any errors / warnings in Xcode itself, with a regular build, and after running the analyzer as well)